core/src/membank.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Robin Linden <dev@robinlinden.eu>
+
+#pragma once
+
+#include "core/imembank.h"
+
+#include <array>
+#include <cstdint>
+#include <utility>
+
+namespace n_e_s::core {
+
+template <uint16_t StartAddr,
+        uint16_t Size = 1u,
```

Comment:
Does it make sense to have default values for these? Should we even provide EndAddr here? It sort of allows us to break things by specifying something like `MemBank<0, 100, 1>`. Might make more sense to either specify just a beginning and end or start and size. I think I prefer beginning and end.

Reply:
Ah, right. :P I didn't get how it was meant to be used. You don't need to add more static_asserts unless you want to, but I think it'd be nicer to remove the default values and add a comment about the mirroring behaviour.